src/main/java/uk/gov/hmcts/reform/sscs/utils/HearingUtils.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.reform.sscs.utils;
+
+import uk.gov.hmcts.reform.sscs.ccd.domain.ExcludeDate;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingOptions;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingSubtype;
+import uk.gov.hmcts.reform.sscs.model.servicehearingvalues.UnavailabilityRange;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public final class HearingUtils {
+    public static final String FACE_TO_FACE = "faceToFace";
+    public static final String TELEPHONE = "telephone";
+    public static final String VIDEO = "video";
+    public static final String PAPER = "paper";
+
+    private HearingUtils() {
+        throw new IllegalStateException("Utility class");
+    }
+
+
+    /**
+     * Return a List of UnavailabilityRange objects, given a HearingOptions object.
+     *
+     * @param hearingOptions HearingOptions object
+     * @return a List of UnavailabilityRange objects
+     */
+    public static List<UnavailabilityRange> getPartyUnavailabilityRange(HearingOptions hearingOptions) {
+        if (Objects.nonNull(hearingOptions.getExcludeDates())) {
+            List<UnavailabilityRange> unavailabilityRanges = new ArrayList<>();
+            for (ExcludeDate excludeDate : hearingOptions.getExcludeDates()) {
+                unavailabilityRanges.add(UnavailabilityRange.builder()
+                        .unavailableFromDate(excludeDate.getValue().getStart())
+                        .unavailableToDate(excludeDate.getValue().getEnd()).build());
+            }
+            return unavailabilityRanges;
+        } else {
+            return new ArrayList<>();
+        }
+    }
```

Comment:
This has already been implemented
https://github.com/hmcts/sscs-hearings-api/blob/master/src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java#L262
It might be better to use that than reimplement it

Reply:
Refactored.
Pretty painful as the change from Strings to LocalDates broke several tests.